### PR TITLE
Add checkoutTokens filter to orders query

### PR DESCRIPTION
**Pull request by Air-t**
I want to merge this change because it adds filtering needed for apps to avoid using a hack (use checkoutComplete mutation to get checkout information). Now apps can filter orders by UUID of checkout.

Fixes: https://linear.app/saleor/issue/SHOPX-567/ir-170-add-api-for-fetching-order-based-on-checkouts-token

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

File: saleor/graphql/order/tests/queries/test_order_with_filter.py [hunk 1]
```diff
@@ -1360,3 +1360,62 @@ def test_order_query_with_filter_by_empty_list(
...
+    staff_api_client,
+    permission_group_manage_orders,
+    orders,
+    checkout,
+    channel_USD,
+):
+    assert not any([order.checkout_token for order in orders])
+    order = orders[0]
+    order.checkout_token = checkout.pk
+    order.save()
```
Comment (IKarbowiak) [inline saleor/graphql/order/tests/queries/test_order_with_filter.py:1376]: You are setting the value only on one object, IMO few orders should have `checkout_token` set to check if the filter is working properly.

File: saleor/graphql/order/tests/queries/test_order_with_filter.py [hunk 2]
```diff
@@ -1360,3 +1360,62 @@ def test_order_query_with_filter_by_empty_list(
...
+    # given
+    variables = {
+        "filter": {
+            "checkoutTokens": [checkout.pk],
```
Comment (IKarbowiak) [inline saleor/graphql/order/tests/queries/test_order_with_filter.py:1380]: We should also have a test for a empty list as `checkooutTokens`

File: saleor/graphql/order/tests/queries/test_order_with_filter.py [hunk 3]
```diff
@@ -1360,3 +1360,62 @@ def test_order_query_with_filter_by_empty_list(
...
+    response = staff_api_client.post_graphql(orders_query_with_filter, variables)
+
+    # then
+    content = get_graphql_content(response)
+    order_data = content["data"]["orders"]["edges"]
+    assert len(order_data) == 1
+    assert {"node": {"id": graphene.Node.to_global_id("Order", order.pk)}} in order_data
+
+
+def test_order_query_with_filter_checkout_tokens_invalid_input(
```
Comment (IKarbowiak) [inline saleor/graphql/order/tests/queries/test_order_with_filter.py:1395]: This test is checking the graphql validation, we do not need such tests

File: saleor/graphql/order/filters.py [hunk 4]
```diff
@@ -20,6 +20,7 @@
     MetadataFilterBase,
     ObjectTypeFilter,
 )
+from ..core.scalars import UUID as CoreUUID
```
Comment (IKarbowiak) [inline saleor/graphql/order/filters.py:23]: Maybe instead:
```suggestion
from ..core.scalars import UUID as UUIDScalar
```

File: saleor/graphql/order/tests/queries/test_order_with_filter.py [hunk 5]
```diff
@@ -1360,3 +1360,66 @@ def test_order_query_with_filter_by_empty_list(
...
+    order,
+    orders_from_checkout,
+    channel_USD,
+):
+    assert not order.checkout_token
+    assert all([order.checkout_token for order in orders_from_checkout])
+    # given
+    variables = {
+        "filter": {
+            "checkoutTokens": [order.checkout_token for order in orders_from_checkout],
```
Comment (IKarbowiak) [inline saleor/graphql/order/tests/queries/test_order_with_filter.py:1378]: Do you think this test will make sure your filter is working properly?
You are providing here the `checkout_token` of all existing orders.

Comment (Air-t) [reply saleor/graphql/order/tests/queries/test_order_with_filter.py:1378]: there is an additional one `order` which is added as a fixture and there are 2 asserts that checks if order.checkout_token is not assigned

**Review by IKarbowiak — APPROVED** [review]

**Review by zedzior — APPROVED** [review]
